cmd/oj/main.go

```diff
@@ -317,6 +329,76 @@ func run() (err error) {
 	return
 }
 
+func digParse(r io.Reader) error {
```

Comment:
I'm sorry but the complexity of this method is insane.

It has to be split

Reply:
I beg to differ. The first part just sets up the callback function picking one according to 3 options so 8 variations. It could be done is a separate function but that would not make it any less complex. I any case just because there are nested if-else block does not make it complex. A switch statement could also be used but then each condition would be evaluated multiple time and at least in my mind it would be less clear what was being done. I did add a comment if that helps.

Reply:
I agree the complexity will be the same, but the code would be easier to read and understand

I would like to suggest you to try splitting code senOut

It appears to me the code would be clearer by having one method for senOut and one for the json (senOut=false)

Reply:
That only moves the condition check for omit and annotate into the functions. For a large file those conditions would be checked on every call instead of once. Having separate functions is preferred. I'm not sure how to make the function selection any more clear. I'll stick with separate functions but if you have an idea on how to make the selection simpler I'd be happy to do that.

Reply:
Unfortunately, I have nothing else to suggest.

Reply:
I'll leave it as is for now. If either of us comes up with an idea it can be changed later.